apps/recovery-portal/src/main/webapp/lite-user-confirm.jsp

```diff
@@ -62,7 +63,16 @@
         CodeValidationRequest validationRequest = new CodeValidationRequest();
         validationRequest.setCode(confirmationKey);
         validationRequest.setProperties(properties);
-        notificationApi.validateCodePostCall(validationRequest);
+        CodeIntrospectResponse codeIntrospectResponse = liteRegisterApi.introspectCode(validationRequest);
+
+        if ((codeIntrospectResponse != null) && (codeIntrospectResponse.getUser() != null)
+                && (codeIntrospectResponse.getUser().getUsername() != null)) {
+            ExtendedUser extendedUser = codeIntrospectResponse.getUser();
+            String username = extendedUser.getUsername();
+            callback = callback + "?confirmation=" + confirmationKey + "&username=" + extendedUser.getUsername();
+        } else {
+            throw new ApiException(400, "Provided confirmation code is invalid.");
```

Comment:
Shall we add the code here in the error

Reply:
Also, would 401 be a better match?  I felt both suits here..

Reply:
Thanks. Addressed.